assets/embedded-files/tekton/staging.yaml

```diff
@@ -1,112 +1,147 @@
-    benefits. This command uses a builder to construct the image, and pushes it to the registry provided.
+
+  workspaces:
+    - name: source
+      description: Directory where application source is located.
+    - name: cache
+      description: Directory where cache is stored (when no cache image is provided).
+      optional: true
+
   params:
+    - name: APP_IMAGE
+      description: The name of where to store the app image.
     - name: BUILDER_IMAGE
       description: The image on which builds will run (must include lifecycle and compatible buildpacks).
-    - name: CACHE
-      description: The name of the persistent app cache volume.
-      default: empty-dir
+    - name: SOURCE_SUBPATH
+      description: A subpath within the `source` input where the source to build is located.
+      default: ""
+    - name: ENV_VARS
+      type: array
+      description: Environment variables to set during _build-time_.
+      default: []
+    - name: PROCESS_TYPE
+      description: The default process type to set on the image.
+      default: "web"
+    - name: RUN_IMAGE
+      description: Reference to a run image to use.
+      default: ""
     - name: CACHE_IMAGE
-      description: The name of the persistent app cache image.
+      description: The name of the persistent app cache image (if no cache workspace is provided).
       default: ""
-    - name: PLATFORM_DIR
-      description: The name of the platform directory.
-      default: empty-dir
+    - name: SKIP_RESTORE
+      description: Do not write layer metadata or restore cached layers.
+      default: "false"
     - name: USER_ID
       description: The user ID of the builder image user.
       default: "1000"
     - name: GROUP_ID
       description: The group ID of the builder image user.
       default: "1000"
-    - name: PROCESS_TYPE
-      description: The default process type to set on the image.
-      default: "web"
-    - name: SOURCE_SUBPATH
-      description: A subpath within the `source` input where the source to build is located.
-      default: ""
-    - name: SKIP_RESTORE
-      description: Do not write layer metadata or restore cached layers
-      default: "false"
-    - name: RUN_IMAGE
-      description: Reference to a run image to use
-      default: ""
+    - name: PLATFORM_DIR
+      description: The name of the platform directory.
+      default: empty-dir
 
-  resources:
-    outputs:
-      - name: image
-        type: image
-
-  workspaces:
-    - name: source
+  results:
+    - name: APP_IMAGE_DIGEST
+      description: The digest of the built `APP_IMAGE`.
 
   stepTemplate:
     env:
       - name: CNB_PLATFORM_API
-        value: "0.3"
+        value: "0.4"
 
   steps:
     - name: prepare
-      # Latest alpine as of Oct 22, 2020
-      image: quay.io/jitesoft/alpine:3.11
-      imagePullPolicy: Always
-      command: ["/bin/sh"]
+      # TODO: Replace with quay.io/jitesoft/alpine:3.11 to avoid dockerhub rate limiting?
```

Comment:
TODO

Reply:
We have other images that come from dockerhub (e.g. epinio-server). I would suggest we stick to the one used in the upstream task until it's a problem. We already have mechanisms to authenticate to dockerhub and get around the issue.